File: ui/src/types/session.ts
/** Maps to internal/domains/auth/models.MySessionItem.
 *  Self-scoped active session of the signed-in user. The backend never exposes
 *  refresh_token / token_id. */
export interface MySessionItem {
	id: string;
	client_ip: string;
	user_agent: string;
	/** RFC3339; empty = unknown. */
	last_login_at: string;
	/** RFC3339; empty = unknown. */
	expires_at: string;
	/** Owning app service id; empty = first-party isme session. */
	app_service_id: string;
	/** Owning app display name; empty for first-party isme or a deleted app. */
	app_name: string;
	/** Owning app stored icon key; empty for first-party isme or a deleted app. */
	app_icon: string;
	/** Owning app stored color key; empty for first-party isme or a deleted app. */
	app_color: string;
	/** True for the session the current request is authenticated with. */
	current: boolean;
	/** Lifetime token-rotation count for this session. */
	refresh_count: number;
	/** RFC3339 of the most recent rotation; empty = never refreshed. */
	last_refreshed_at: string;
}

/** Maps to internal/domains/auth/models.MySessionCount. */
export interface MySessionCount {
	count: number;
	new_in_24h: number;
	/** Accurate sliding-24h token-rotation count for the user. */
	rotations_24h: number;
	/** RFC3339 of the user's most recent rotation; empty = no refreshes yet. */
	last_refreshed_at: string;
}

File: internal/domains/auth/models/session.go
package models

// MySessionItem is a single active session belonging to the current user.
// It deliberately omits RefreshToken and TokenID so they are never exposed
// over the API.
type MySessionItem struct {
	ID           string `json:"id"`
	ClientIP     string `json:"client_ip"`
	UserAgent    string `json:"user_agent"`
	LastLoginAt  string `json:"last_login_at"`
	ExpiresAt    string `json:"expires_at"`
	AppServiceID string `json:"app_service_id"`
	// AppName/AppIcon/AppColor are enriched from the owning app_service for
	// SSO sessions. They are empty for first-party isme sessions (empty
	// AppServiceID) and for sessions whose app_service has been deleted.
	AppName  string `json:"app_name"`
	AppIcon  string `json:"app_icon"`
	AppColor string `json:"app_color"`
	Current  bool   `json:"current"`
	// RefreshCount is the lifetime token-rotation count for this session.
	RefreshCount int64 `json:"refresh_count"`
	// LastRefreshedAt is RFC3339 of the most recent rotation; "" = never refreshed.
	LastRefreshedAt string `json:"last_refreshed_at"`
}

// MySessionCount is the active-session summary used by the Welcome stat cards.
type MySessionCount struct {
	Count    int `json:"count"`
	NewIn24h int `json:"new_in_24h"`
	// Rotations24h is the accurate sliding-24h token-rotation count for the user,
	// derived from token_rotation_events (drives the "Token rotations" card).
	Rotations24h int `json:"rotations_24h"`
	// LastRefreshedAt is RFC3339 of the user's most recent rotation across active
	// sessions; "" = no refreshes yet.
	LastRefreshedAt string `json:"last_refreshed_at"`
}
